pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1986,52 +1986,90 @@ cdef class YearBegin(YearOffset):
 # Quarter-Based Offset Classes
 
 cdef class QuarterOffset(SingleConstructorOffset):
-    _attributes = tuple(["n", "normalize", "startingMonth"])
+    """
+    Base for quarter-based offset classes.
+
+    Parameters
+    ----------
+    n : int, default 1
+        Number of quarters to offset.
+    normalize : bool, default False
+        If true, the time component of the resulting date-time is converted
+        to 00:00:00, i.e. midnight (the start, not the end of date-time).
+    month : int
+        The calendar month number (1 for January) of the beginning or ending
+        month in a custom-defined quarter to be used as an offset.
+
+        .. versionadded:: 1.X.X
+    startingMonth : int
+        The calendar month number (1 for January) of the beginning or ending
+        month in a custom-defined quarter to be used as an offset.
+
+        .. deprecated:: 1.X.X
+    """
+
+    _attributes = tuple(["n", "normalize", "month"])
     # TODO: Consider combining QuarterOffset and YearOffset __init__ at some
-    #       point.  Also apply_index, is_on_offset, rule_code if
-    #       startingMonth vs month attr names are resolved
+    #       point. Also apply_index, is_on_offset, rule_code.
 
-    # FIXME: python annotations here breaks things
-    # _default_starting_month: int
-    # _from_name_starting_month: int
+    # FIXME: python annotation here breaks things
 
     cdef readonly:
-        int startingMonth
+        int month
+        int startingMonth  # Backwards compatibility
 
-    def __init__(self, n=1, normalize=False, startingMonth=None):
+    def __init__(self, n=1, normalize=False, month=None,
+                 *, startingMonth=None):  # Backwards compatibility
         BaseOffset.__init__(self, n, normalize)
 
-        if startingMonth is None:
-            startingMonth = self._default_starting_month
-        self.startingMonth = startingMonth
+        # Backwards compatibility
+        if month and startingMonth and month != startingMonth:
+            raise TypeError("Offset received both month and startingMonth")
+        elif month is None and startingMonth is not None:
+            warnings.warn(
+                "startingMonth is deprecated, use month instead.",
+                DeprecationWarning,
+                stacklevel=2
+            )
+            month = startingMonth
+
+        month = month if month is not None else self._default_month
+        self.month = month
+
+        if month < 1 or month > 12:
+            raise ValueError("Month must go from 1 to 12.")
 
     cpdef __setstate__(self, state):
-        self.startingMonth = state.pop("startingMonth")
+        try:
+            self.month = state.pop("month")
+        except:
```

Comment:
can you be more specifi on the except

Reply:
I added a longer comment there.

Reply:
a) im pretty sure jreback meant not to use a bare "except:", but to specifically catch `KeyError`

b) rule of thumb: pls let the person who asked the question hit the "resolve conversation" button; something might be clear to you but not to them

Reply:
yeah this is non-obvious on the exception.